components/bin/build

```diff
@@ -0,0 +1,318 @@
+const LIB = config.lib || './lib';                                          // path to the lib directory to create
+const COMPONENT = config.component || 'part';                               // name of the component
+const GLOBAL = config.global || `../${MATHJAX3}/components/global.js`;      // the location of global.js
+
+/**
+ * The list of files that need to be added to the lib directory
+ */
+let PACKAGE = [];
+
+/**
+ * Process an array of files/directories to be included in the component
+ *
+ * @param {string} base    The root directory for the .ts files
+ * @param {string} dir     The relative path within the root for the files to process
+ * @param {string[]} list  An array of file/directory names to process
+ */
+function processList(base, dir, list) {
+    for (const item of list) {
+        const file = path.join(dir, item);
+        if (!exclude.has(file)) {
+            const stat = fs.statSync(path.resolve(base, file));
+            if (stat.isDirectory()) {
+                processDir(base, file);
+            } else if (file.match(/\.ts/)) {
+                processFile(base, file);
+            }
+        }
+    }
+}
+
+/**
+ * Process the contents of the directory
+ *
+ * @param {string} base    The root directory for the .ts files
+ * @param {string} dir     The relative path within the root for the directory to process
+ */
+function processDir(base, dir) {
+    const root = path.resolve(base, dir);
+    processList(base, dir, fs.readdirSync(root));
+}
+
+/**
+ * Process the contents of a .ts file
+ *
+ * Look for export commands within the file, and determine the objects
+ *   that they reference, then produce the library file that defines
+ *   them, and add the file to the package list.
+ *
+ * @param {string} base    The root directory for the .ts files
+ * @param {string} name    The path of the file relative to the base
+ */
+function processFile(base, name) {
+    console.log(name);
+    const file = fs.readFileSync(path.resolve(base, name)).toString();
+    const parts =
+        file.split(/(^export(?:\s+default)?(?:\s+abstract)?\s+[^ {*}]+\s+[a-zA-Z0-9_.$]+)/m);
+    const objects = processParts(parts);
+    const lines = processLines(name, objects);
+    makeFile(name, lines);
+    if (objects.length) {
+        PACKAGE.push(name);
+    }
+}
+
+/**
+ * Process the export lines to record the ones that need to be added to the
+ *   library file.
+ *
+ * @param {string[]} parts       The file broken into export lines and the text between them
+ * @return {string[]}            An array of names of exported objects
+ */
+function processParts(parts) {
+    const objects = [];
+    for (let i = 1; i < parts.length; i += 2) {
+        const words = parts[i].split(/\s+/);
+        const type = words[words.length - 2];
+        const name = words[words.length - 1];
+
+        if (words[1] === 'default') {
+
+            objects.push('default');
+
+        } else {
+
+            switch (type) {
+              case 'type':
```

Comment:
Same here: We need to allow for changes in syntax. Maybe via a dictionary, rather than a switch.